### PR TITLE
Add flatsome compatibility for RUCSS inline CSS preserve

### PR DESCRIPTION
**Pull request by remyperona**
## Description

Add a compatibility file for Flatsome, to add specific patterns to the inline CSS content to preserve when using remove unused CSS

Fixes https://github.com/wp-media/nodejs-treeshaker/issues/39

## Type of change

- [x] Third-Party compatibility

# Checklist:

Please delete the options that are not relevant.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by mostafa-hisham — APPROVED** [review]

**Comment by Mai-Saad** [comment]
@Tabrisrp currently there is a missing section, styles are different, and the grey background for contact is not there. can you please check?

File: inc/ThirdParty/Themes/Flatsome.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	 * Checks if the current theme (or parent) is Flatsome
+	 *
+	 * @since 3.11
+	 *
+	 * @param WP_Theme $theme Instance of the theme.
+	 */
+	private static function is_flatsome( $theme = null ) {
+		$theme = $theme instanceof \WP_Theme ? $theme : wp_get_theme();
+
+		return 'flatsome' === ( strtolower( $theme->get( 'Name' ) ) || strtolower( $theme->get_template() ) );
```
Comment (mostafa-hisham) [inline inc/ThirdParty/Themes/Flatsome.php]: the problem was in condition
```suggestion
		return 'flatsome' === ( strtolower( $theme->get( 'Name' ) ) ?: strtolower( $theme->get_template() ) );
```
OR
```suggestion
		return 'flatsome' ===  strtolower( $theme->get( 'Name' ) )
                || 'flatsome' ===  strtolower( $theme->get_template() );
```

**Review by mostafa-hisham — CHANGES_REQUESTED** [review]

**Review by mostafa-hisham — APPROVED** [review]

**Comment by Mai-Saad** [comment]
The missing section is added now, the styles are the same and grey background is added for contact.

The only difference now compared to ?nowprocket is the spacing above the contact section. Can we fix this point?

**Comment by engahmeds3ed** [comment]
this spacing is happening from the following `gap-timestamp` id as follows:-

?nowprocket version
image

The cached version
image

So we need to add `#gap-` into the list of inline preserve list.

**Review by Mai-Saad — APPROVED** [review]
Working as expected on both Flatsome and ChildFlatsome

**Pull request merged**